src/helper.js

```diff
@@ -170,7 +170,32 @@ export const cloneWithProject = (sourceDm, projField, config, allFields) => {
     return cloned;
 };
 
+const sanitizeSchema = (schema) => {
+    // Do deep clone of the schema as the user might change the input schema reference.
+    const clonedSchema = JSON.parse(JSON.stringify(schema));
+    return clonedSchema.map((s) => {
```

Comment:
change the name of the function parameter so that it is readable. Something like `s` to `unitSchema`.